Null-safety migration and bug fixes

caseyryan opened the pull request:
- Migrated to null-safety 
- Updated gradle and android sdks versions
- Ported to Flutter embedding 2
- Fixed a bug which caused the next error: "Attempted to finish an input event but the input event receiver has already been disposed" when tapping on a screen when local auth has been running on Android
- Added a max delay timeout which will kill local auth after some time of inactivity. This is made to avoid memory leaks

priezz commented on android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java:
```diff
@@ -123,19 +125,19 @@ private void start() {
     resume();
   }
 
+  private void print(Object value) {
```
please rename to `logDebug`

priezz commented on android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java:
```diff
@@ -94,7 +94,9 @@
   }
 
   void authenticate() {
+    Log.d("AuthenticationHelper", "Start authentication");
```
Use logDebug() call instead?

priezz reviewed (changes requested):
@caseyryan Konstantin, thank you for your contribution! 

Please
- Remove `android/.idea` and `android/.settings` directories and `android/local.properties`, `android/app/.classpath` and `android/app/.project` files from the PR (from both the package code and the example)
- Update iOS code from the upstream

I'll test the package after these changes.

caseyryan commented:
I've removed android studio files and renamed print to logDebug. Sorry I don't have time to look into ios part. As far as I know iOS worked without any bugs .

priezz commented:
> I've removed android studio files and renamed print to logDebug. Sorry I don't have time to look into ios part. As far as I know iOS worked without any bugs .

iOS - ok. Please remove all the other local artefacts, i.e. `android/.gradle`, `android/.settings/org.eclipse.buildship.core.prefs`, `android/local.properties`, `android/app/.classpath`, `android/app/.project` - from both, the root of the project and from the `example`.

caseyryan commented:
> > I've removed android studio files and renamed print to logDebug. Sorry I don't have time to look into ios part. As far as I know iOS worked without any bugs .
> 
> iOS - ok. Please remove all the other local artefacts, i.e. `android/.gradle`, `android/.settings/org.eclipse.buildship.core.prefs`, `android/local.properties`, `android/app/.classpath`, `android/app/.project` - from both, the root of the project and from the `example`.

They are all removed already

priezz commented:
> They are all removed already and added to .gitignore

Not all of them, please look carefully at the list above and [here](https://github.com/priezz/flutter_local_auth_invisible/pull/15/files) at the actual files list in the PR.

caseyryan commented:
I don't understand what you're talking about. I've already spent too much time fixing a plugin that's not even mine. If you don't what to merge it or fix it, that's your right, I can use it directly from my repo

priezz commented:
I have clearly listed the files, which are to be removed. Especially the `android/.gradle` directories. Your PR brings **90** changed/updated files in the repository while only 20 of them contain the changes and all the others are you local build or IDE artefacts.

caseyryan commented:
I just wanted to help you update your plugin. I don't really use Android Studio so I have no idea of what it might add to the project. Since those files and directories were not in your .gitignore they got into the project. I removed some of them but looking for others takes a lot of time which I don't have. So if you don't care for the project and don't want to keep it up to date, I don't as well. All I needed I've made in my fork and it works for me. Have a good day
